src/components/dropdowns/base/index.js

```diff
@@ -0,0 +1,50 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import { TextField, MenuItem } from '@material-ui/core';
+import { TextField as FormikTextField } from 'formik-material-ui';
+import { Field } from 'formik';
+
+const GenericDropdownField = props => {
+  const { name, label, children, readOnly, ...otherProps } = props;
+
+  if (readOnly) {
+    const readOnlyProps = {
+      disableUnderline: true,
+      readOnly: true,
+    };
+
+    return (
+      <TextField label={label} InputProps={readOnlyProps} {...otherProps} />
+    );
+  }
+
+  return (
+    <Field
+      component={FormikTextField}
+      select
+      name={name}
+      label={label}
+      {...otherProps}
+    >
+      {React.Children.map(children, child => (
```

Comment:
not sure why this is children, maybe selection as props as array of words is better?

Reply:
Either way works here.

Reply:
right - just pointing out that props.selections.map is a little more idiomatic than React.Children.map

Reply:
I can change it to that. To be frank we then won't have to worry about props.selections's type since we can enforce it to be an array.

Reply:
exactly!